debian/postinst

```diff
@@ -67,8 +68,31 @@ upgrade_to_status_cache() {
     ua status 2>&1 > /dev/null
 }
 
+is_release_upgrade() {
+    # Return 0 if the package version we upgrade from does not match the
+    # current series (e.g. 14.04)
+    if test "${2#*$VERSION_ID}" = ${2}; then
+        # Our current VERSION_ID from os-release is not the same as previous
+        # package
+        return 0
+    else
+        # previous pkg version contains the same VERSION_ID as upgraded pkg ver
+        return 1
+    fi
+}
+
 case "$1" in
     configure)
+      # Special case: legacy precise creds allowed for trusty esm
+      # do-release-upgrade would have properly s/precise/trusty/ in apt source
+      # Issue: #693
+      if is_release_upgrade; then
+          if test "${2#*12.04}" != ${2}; then   # then upgraded from precise
```

Comment:
Is it worth considering a `PREVIOUS_VERSION_RELEASE` variable (or one with a better name :grin:) that captures the value once and can be used in both places where we currently have this logic?

Reply:
done